internal/pkg/policy/policy_output.go

```diff
@@ -246,6 +271,71 @@ func (p *Output) prepareElasticsearch(
 	return nil
 }
 
+func (p *Output) createAndGetBulker(ctx context.Context, mainBulker bulk.Bulk, outputMap map[string]map[string]interface{}) (bulk.Bulk, error) {
+	bulker := bulkerMap[p.Name]
+	if bulker != nil {
+		// TODO replace if hosts or service token changed?
+		return bulker, nil
+	}
+	bulkCtx, bulkCancel := context.WithCancel(context.Background())
+	defer bulkCancel()
+	es, err := p.createRemoteEsClient(bulkCtx, outputMap)
+	if err != nil {
+		return nil, err
+	}
+	// starting a new bulker to create/update API keys for remote ES output
+	newBulker := bulk.NewBulker(es, nil, mainBulker.Opts())
```

Comment:
this code is mostly copying how a bulker is started in https://github.com/elastic/fleet-server/blob/475e3ee0acdc0b9e33eee60cea1e5cabe91ced64/internal/pkg/server/fleet.go#L363-L417, without the extra steps like `runSubsystems`.

Reply:
As I commented before, we should be able to use `bulk.NewBulker(es, tracer)`.

Also we should set the tracer to the one that fleet-server uses.

Reply:
updated